### ISSUE TITLE
Downloads fail when Transfer-Encoding = chunked

### ISSUE THREAD

**Issue by josh4trunks** [nextcloud/android#113]
### Actual behaviour
- Download failed - Download of test.txt could not be completed
### Expected behaviour
- File should be downloaded successfully
### Steps to reproduce
1. Setup server environment where files will be served with `Transfer-Encoding = chunked`
2. Try downloading file using android app
### Environment data

Nextcloud app version:
1.1.0
#### 

This bug is also present is the ownCloud android app. https://github.com/owncloud/android/issues/1128
The fix seems to be discussed breifly there, though I don't see any commits referenced.

**Comment by tobiasKaminsky** [comment]
Can you help me and and explain where and what to set in the server environment to reproduce this behaviour?

**Comment by josh4trunks** [comment]
In my case chunking happens because I have Varnish in front of my nginx webserver compressing text files with gzip. I don't know how to reproduce this directly but maybe people in the owncloud bug know. possibly by enabling gzip chunked is used?

**Comment by tobiasKaminsky** [comment]
I thought it is a supported server option...
With this I doubt that we can support this.

**Comment by josh4trunks** [comment]
I believe there is a way to have chunked encoding without gzip enabled. It is part of the HTTP/1.1 specification.
But we can just wait till this is fixed in the ownCloud android client and apply the same changes.

**Comment by johnwmail** [comment]
> Can you help me and and explain where and what to set in the server environment to reproduce this behaviour?

Hi, OpenBSD's httpd is enable/force chunked encoding by default.

**Comment by johnwmail** [comment]
I still have this problem, how to solve it.
Any workaround?

**Comment by josh4trunks** [comment]
In my environment, where this problem was caused by some type of incompatibility with Varnish, I was able to get this working by setting `beresp.do_stream = false` whenever I enable gzip for `content-type = text/application`.

Maybe you can check what Varnish is doing wrong/weird with do_stream

**Comment by johnwmail** [comment]
So, still no fix on client side?

**Comment by mario** [comment]
Does this happens only on 3/4g or also on wifi?

**Comment by mario** [comment]
Also, can you inspect the headers and tell me what does it says for content-size?

**Comment by johnwmail** [comment]
Yes, both 3/4g and wifi also have this problem.

I don't know how to inspect the header, this is https, tcpdump not work, I will try to Google it later

Thanks

**Comment by johnwmail** [comment]
Hello, any idea how to fix it?

**Comment by tobiasKaminsky** [comment]
@josh4trunks can you send us also credentials for a test account to android at nextcloud dot com?
Then I can have a look, but as oC also does not have a solution, I am unsure if we can do it...

**Comment by johnwmail** [comment]
@tobiasKaminsky  I would also setup test account (with Transfer-Encoding = chunked) for this.
But I do not want to post login/password here, would you send email to me?

johnw.mail at gmail dot com

Thanks.

**Comment by josh4trunks** [comment]
@tobiasKaminsky I already fixed this in my production environment. See https://github.com/nextcloud/android/issues/113#issuecomment-288131898

Did you still want an account on my working Nextcloud install?

**Comment by tobiasKaminsky** [comment]
Ahh, I overread that you have solved it and that is (more or less) a server/environment side problem.
So I will close this.

**Comment by mario** [comment]
Please re-open. This has to be solved client side as well.

**Comment by johnwmail** [comment]
@mario agree, this should be solved client side as well.
I tried talk to OpenBSD people, 

http://marc.info/?l=openbsd-misc&m=149422158413997&w=2

They quoted RFC7230

"A recipient MUST be able to parse and decode the chunked transfer coding."

I would like to setup nextcloud AC with "chunked transfer coding" enabled.

Email me, if people who want/need the AC to debug/testing

johnw dot mail at gmail dot com

Thanks.

**Comment by johnwmail** [comment]
https://tools.ietf.org/html/rfc7230#section-4.1

**Comment by tobiasKaminsky** [comment]
A quick search if our http library jackrabbit (http://jackrabbit.apache.org/jcr/components/jackrabbit-webdav-library.html) is supporting this does not yield anything...
So if you can give me a starting point I will give it a try.

**Comment by tdm4** [comment]
Having the same problem here. Using nginx instead of httpd for now (on OpenBSD) with ```chunked_transfer_encoding off;```
Not ideal, but it works. Android client breaks RFC7230 unfortunately. :(

**Comment by johnwmail** [comment]
Hi @tdm4

I written a dirty hack to disable chunked_transfer_encoding for OpenBSD httpd,
Maybe you will interesting

http://marc.info/?l=openbsd-misc&m=149422389314402&w=2

**Comment by rpcameron** [comment]
Is there any activity on this, or are the Nextcloud devs going to wait on Owncloud to fix this since the download function/code seems to be in their Android library code?

Until this issue is resolved, I cannot use either client. Yes, I see the patch for OpenBSD httpd, but I'm not going to modify my system software to fix a bug that exists in other third-party software because the third-party software is broken and doesn't properly implement their HTTP client.

**Comment by mario** [comment]
Actually our library differs from theirs in quite a lot of things...

But yea, hopefully soon...

**Comment by rpcameron** [comment]
@mario Ah, sorry. When I was trying to sort through the code I must have followed a wrong reference when tracking source files and includes. Does this mean that Nextcloud will soon finally support this, so I no longer have field support problems from users using Android devices? Or will they still need to be redirected to some other app to access their files on their phones ...

**Comment by johnwmail** [comment]
@rpcameron Is it owncloud, already fixed this bug?

**Comment by rpcameron** [comment]
@johnwmail No, it is still open there, too (1128, since 31 Aug 2015). It has been acknowledged, constantly postponed, and now is "backlogged".

**Comment by edermi** [comment]
Same issue here. Would love to access the nextcloud hosted on my OpenBSD with the Android App

**Comment by rpcameron** [comment]
Is there any movement on this bug? I recently saw that version 2 RCs are available so I decided to give them a shot, but no luck. It almost seems there's no interest from the developers to fix this.

(And yes, I know the old adage about fixing it yourself if you want it done. I have investigated it to try and find the relevant code, but I haven't programmed in Java in over a decade, and trying to find the relevant functions left me in a knot bouncing from class to class across the different repos.)

**Comment by tobiasKaminsky** [comment]
I tried it, but it is not that easy to fix. 
I can have a look again once I have a little more time :-/

**Comment by rpcameron** [comment]
So, 2.0 was just released, but this bug that's been present since the beginning has no movement.

This app is still useless for accessing files on Android if chunked downloads are not supported as per the HTTP spec.

**Comment by xstasi** [comment]
Hi there, poking in to say that I too am afflicted by this issue.

+1

**Comment by tobiasKaminsky** [comment]
Can someone provide an test account on a server where this is happening?
Please send it to android at nextcloud dot com with a reference to this issue.

**Comment by rpcameron** [comment]
@tobiasKaminsky I just sent an email with login details for a test account on an OpenBSD 6.2 server running its httpd, which only serves with "transfer encoding = chunked". Maybe this means he Android client will finally support the HTTP spec ...

**Comment by tobiasKaminsky** [comment]
@rpcameron thank you. I do my very best :-)

**Comment by tobiasKaminsky** [comment]
@xstasi thank you also for the credentials 👍

**Comment by tobiasKaminsky** [comment]
I guess I have a fix for this: https://github.com/nextcloud/android-library/pull/98

We will soon have our dev version back again. This builds a nearly nightly snapshot from master. So once the fix is merged, you can give it a try and report back :-)

**Comment by rpcameron** [comment]
Just ran a quick test with the latest Dev version (20171204), and it appears to be downloaded files (at long last). Thanks to the devs for finally addressing this issue ... Hopefully a stable release will soon make its way out.

**Comment by AndyScherzinger** [comment]
https://github.com/nextcloud/android-library/pull/98 did fix it and has been integrated with master and with the new dev builds now been shipped for the dev version. The next stable version doesn't have a date yet but is also progressing quite well: https://github.com/nextcloud/android/milestones/Nextcloud%20App%202.1.0

**Comment by johnwmail** [comment]
Where can we download the dev app?

**Comment by AndyScherzinger** [comment]
@johnwmail it is distributed via f-droid: https://f-droid.org/packages/com.nextcloud.android.beta/ (so best is to install the f-droid client and Search for "Nextcloud Dev").

**Comment by tobiasKaminsky** [comment]
Please note that fdroid currently does not have the latest version, it being built this day and will be available soon.

**Comment by levaidaniel** [comment]
Hi there!

Is it normal that version 3.5.1 (March, 18, 2019) still doesn't download anything from OpenBSD httpd(8)? Is this mentioned fix supposed to be in this version?

**Comment by johnwmail** [comment]
Hi, they fixed a year.

**Comment by tobiasKaminsky** [comment]
> Is it normal that version 3.5.1 (March, 18, 2019) still doesn't download anything from OpenBSD httpd(8)? Is this mentioned fix supposed to be in this version?

Can you create us a test account, test if the problem occurs also there and if so send the credentials to tobias at nextcloud dot com with a reference to this issue?

**Comment by levaidaniel** [comment]
Sure thing, I sent the mail containing the info.

**Comment by levaidaniel** [comment]
In the meantime, here are the logs from the application and Chrome, respectively. Maybe these shed some light into the problem:
`
"GET /nextcloud/index.php/204 HTTP/1.1" 204 0
"GET /nextcloud/index.php/avatar/username/448 HTTP/1.1" 304 0
"GET /nextcloud/ocs/v1.php/cloud/users/username?format=json HTTP/1.1" 200 0
"GET /nextcloud/ocs/v1.php/cloud/users/username?format=json HTTP/1.1" 200 0
"GET /nextcloud/index.php/204 HTTP/1.1" 204 0
"GET /nextcloud/index.php/204 HTTP/1.1" 204 0
"GET /nextcloud/index.php/204 HTTP/1.1" 204 0
"GET /nextcloud/index.php/204 HTTP/1.1" 204 0
"GET /nextcloud/index.php/204 HTTP/1.1" 204 0
"GET /nextcloud/index.php/204 HTTP/1.1" 204 0
"GET /nextcloud/index.php/avatar/username/448 HTTP/1.1" 304 0
"GET /nextcloud/ocs/v1.php/cloud/users/username?format=json HTTP/1.1" 200 0
"GET /nextcloud/status.php HTTP/1.1" 200 0
"GET /nextcloud/ocs/v1.php/cloud/capabilities?format=json HTTP/1.1" 200 0
"GET /nextcloud/ocs/v1.php/cloud/user?format=json HTTP/1.1" 200 0
"PROPFIND /nextcloud/remote.php/webdav/ HTTP/1.1" 207 0
"PROPFIND /nextcloud/remote.php/webdav/ HTTP/1.1" 207 0
"GET /nextcloud/ocs/v2.php/apps/files_sharing/api/v1/shares?path=%2F&reshares=true&subfiles=true HTTP/1.1" 200 0
"GET /nextcloud/status.php HTTP/1.1" 200 0
"GET /nextcloud/ocs/v1.php/cloud/capabilities?format=json HTTP/1.1" 200 0
"GET /nextcloud/ocs/v1.php/cloud/user?format=json HTTP/1.1" 200 0
"PROPFIND /nextcloud/remote.php/webdav/ HTTP/1.1" 207 0
"PROPFIND /nextcloud/remote.php/webdav/ HTTP/1.1" 207 0
"GET /nextcloud/ocs/v2.php/apps/files_sharing/api/v1/shares?path=%2F&reshares=true&subfiles=true HTTP/1.1" 200 0
"PROPFIND /nextcloud/remote.php/webdav/Zagar/ HTTP/1.1" 207 0
"PROPFIND /nextcloud/remote.php/webdav/Zagar/ HTTP/1.1" 207 0
"GET /nextcloud/remote.php/webdav/Zagar/7FCB0045D6134446B4654308EC246F9C_1_20-00-00-000.pdf HTTP/1.1" 200 0
"GET /nextcloud/ocs/v2.php/apps/files_sharing/api/v1/shares?path=%2FZagar%2F&reshares=true&subfiles=true HTTP/1.1" 200 0
"GET /nextcloud/ocs/v2.php/apps/activity/api/v2/activity/filter?format=json&previews=true&sort=desc&object_type=files&object_id=212 HTTP/1.1" 200 0
"GET /nextcloud/ocs/v2.php/apps/activity/api/v2/activity/filter?format=json&previews=true&sort=desc&object_type=files&object_id=212 HTTP/1.1" 200 0
"PROPFIND /nextcloud/remote.php/dav/versions/username/versions/212 HTTP/1.1" 207 0
"GET /nextcloud/ocs/v2.php/apps/activity/api/v2/activity/filter?format=json&previews=true&sort=desc&object_type=files&object_id=212 HTTP/1.1" 200 0
"PROPFIND /nextcloud/remote.php/dav/versions/username/versions/212 HTTP/1.1" 207 0
"PROPFIND /nextcloud/remote.php/dav/versions/username/versions/212 HTTP/1.1" 207 0
"<UNKNOWN> " 408 0
"<UNKNOWN> " 408 0
"<UNKNOWN> " 408 0
"<UNKNOWN> " 408 0
"<UNKNOWN> " 408 0
`

`
"POST /nextcloud/index.php/login HTTP/1.1" 303 0
"GET /nextcloud/index.php/apps/files/ HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/core/8aae-5e6c-jquery-ui-fixes.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/core/8aae-5e6c-server.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/firstrunwizard/0d6d-5e6c-firstrunwizard.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/core/8aae-5e6c-css-variables.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/notifications/70e2-5e6c-styles.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/files/e8ed-5e6c-merged.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/core/8aae-5e6c-results.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/activity/15ac-5e6c-style.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/core/8aae-5e6c-jquery.ocdialog.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/apps/gallery/css/slideshow.css?v=eaecf245-0 HTTP/1.1" 200 3930
"GET /nextcloud/apps/gallery/css/gallerybutton.css?v=eaecf245-0 HTTP/1.1" 200 2733
"GET /nextcloud/apps/gallery/css/share.css?v=eaecf245-0 HTTP/1.1" 200 3640
"GET /nextcloud/index.php/css/files_trashbin/35c3-5e6c-trash.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/comments/35c3-5e6c-comments.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/comments/35c3-5e6c-autocomplete.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/apps/notifications/js/notifications.js?v=f361828e-0 HTTP/1.1" 200 132709
"GET /nextcloud/index.php/css/core/8aae-5e6c-systemtags.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/apps/files_sharing/js/app.js?v=f361828e-0 HTTP/1.1" 200 9526
"GET /nextcloud/apps/files_sharing/js/sharedfilelist.js?v=f361828e-0 HTTP/1.1" 200 14332
"GET /nextcloud/index.php/css/files_texteditor/b1d7-5e6c-merged.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/apps/files_trashbin/js/app.js?v=f361828e-0 HTTP/1.1" 200 4059
"GET /nextcloud/apps/files_trashbin/js/filelist.js?v=f361828e-0 HTTP/1.1" 200 8863
"GET /nextcloud/index.php/css/icons/icons-vars.css?v=1554124867 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/files_sharing/6a15-5e6c-mergedAdditionalStyles.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/css/systemtags/35c3-5e6c-systemtagsfilelist.css?v=08343dd3f787291558383b46e1b489fb-f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/apps/federatedfilesharing/js/external.js?v=f361828e-0 HTTP/1.1" 200 4936
"GET /nextcloud/core/js/oc-backbone-webdav.js?v=f361828e-0 HTTP/1.1" 200 9565
"GET /nextcloud/index.php/js/files/merged-index.js?v=f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/js/activity/activity-sidebar.js?v=f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/js/files_versions/merged.js?v=f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/apps/notifications/img/notifications.svg HTTP/1.1" 200 456
"GET /nextcloud/index.php/js/files_sharing/additionalScripts.js?v=f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/js/comments/merged.js?v=f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/js/files_texteditor/merged.js?v=f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/settings/img/admin.svg?v=f361828e HTTP/1.1" 200 236
"GET /nextcloud/apps/firstrunwizard/img/info.svg?v=f361828e HTTP/1.1" 200 529
"GET /nextcloud/ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.1" 200 0
"GET /nextcloud/settings/img/help.svg?v=f361828e HTTP/1.1" 200 519
"GET /nextcloud/core/img/actions/logout.svg?v=f361828e HTTP/1.1" 200 342
"GET /nextcloud/core/img/actions/delete.svg HTTP/1.1" 200 247
"GET /nextcloud/index.php/js/core/merged.js?v=f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/apps/files/img/app.svg?v=f361828e HTTP/1.1" 200 222
"GET /nextcloud/apps/activity/img/activity.svg?v=f361828e HTTP/1.1" 200 217
"GET /nextcloud/apps/gallery/img/app.svg?v=f361828e HTTP/1.1" 200 570
"GET /nextcloud/index.php/avatar/username/32?v=1 HTTP/1.1" 201 0
"GET /nextcloud/index.php/js/systemtags/merged.js?v=f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/js/gallery/scripts-for-file-app.js?v=f361828e-0 HTTP/1.1" 200 0
"GET /nextcloud/core/search/templates/part.results.html HTTP/1.1" 200 220
"GET /nextcloud/core/img/breadcrumb.svg?v=1 HTTP/1.1" 200 188
"GET /nextcloud/cron.php HTTP/1.1" 200 0
"PROPFIND /nextcloud/remote.php/dav/files/username/ HTTP/1.1" 207 0
"GET /nextcloud/index.php/apps/gallery/config?extramediatypes=1 HTTP/1.1" 200 0
"GET /nextcloud/index.php/apps/systemtags/lastused HTTP/1.1" 200 0
"GET /nextcloud/ocs/v2.php/core/whatsnew?format=json HTTP/1.1" 204 0
"GET /nextcloud/index.php/apps/theming/img/core/filetypes/folder.svg?v=0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/apps/files/ajax/getstoragestats.php?dir=%2F HTTP/1.1" 200 0
"PROPFIND /nextcloud/remote.php/dav/files/username/Zagar HTTP/1.1" 207 0
"GET /nextcloud/index.php/apps/files/ajax/getstoragestats.php?dir=%2FZagar HTTP/1.1" 200 0
"GET /nextcloud/index.php/apps/theming/img/core/filetypes/application-pdf.svg?v=0 HTTP/1.1" 200 0
"GET /nextcloud/index.php/apps/files_pdfviewer/?file=%2Fnextcloud%2Fremote.php%2Fwebdav%2FZagar%2F7FCB0045D6134446B4654308EC246F9C_1_20-00-00-000.pdf HTTP/1.1" 200 0
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/viewer.css?v=1.4.0 HTTP/1.1" 200 51850
"GET /nextcloud/apps/files_pdfviewer/css/viewer.css?v=1.4.0 HTTP/1.1" 200 519
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/build/pdf.js?v=1.4.0 HTTP/1.1" 200 513649
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/viewer.js?v=1.4.0 HTTP/1.1" 200 350254
"GET /nextcloud/apps/files_pdfviewer/js/workersrc.js?v=1.4.0 HTTP/1.1" 200 3360
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/locale/locale.properties?v=1.4.0 HTTP/1.1" 200 4314
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/texture.png HTTP/1.1" 200 2417
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-viewThumbnail.png HTTP/1.1" 200 185
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-viewAttachments.png HTTP/1.1" 200 384
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-viewOutline.png HTTP/1.1" 200 178
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-sidebarToggle.png HTTP/1.1" 200 225
"GET /nextcloud/apps/files_pdfviewer/img/toolbarButton-secondaryToolbarClose.svg HTTP/1.1" 200 332
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-bookmark.png HTTP/1.1" 200 174
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-zoomIn.png HTTP/1.1" 200 136
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-zoomOut.png HTTP/1.1" 200 88
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-menuArrows.png HTTP/1.1" 200 107
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-pageDown.png HTTP/1.1" 200 238
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-pageUp.png HTTP/1.1" 200 246
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-print.png HTTP/1.1" 200 257
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-presentationMode.png HTTP/1.1" 200 321
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-secondaryToolbarToggle.png HTTP/1.1" 200 243
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-search.png HTTP/1.1" 200 309
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/toolbarButton-download.png HTTP/1.1" 200 259
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/locale/en-GB/viewer.properties HTTP/1.1" 200 7535
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/build/pdf.worker.js?v=1.4.0 HTTP/1.1" 200 1424568
"GET /nextcloud/remote.php/webdav/Zagar/7FCB0045D6134446B4654308EC246F9C_1_20-00-00-000.pdf HTTP/1.1" 200 0
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/shadow.png HTTP/1.1" 200 290
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/loading-icon.gif HTTP/1.1" 200 2545
"GET /nextcloud/apps/files_pdfviewer/vendor/pdfjs/web/images/loading-small.png HTTP/1.1" 200 7402
`

**Comment by levaidaniel** [comment]
Interesting thin is using the same app on Android 7.1.1 yields more or less positive results over using the same v3.5.1 app on Android 9. The former only fails sometimes, usually after waiting for a while after a successful download -- it says the file couldn't be synced so using the downloaded version (or something along these lines).
But other than that, on older Android the success rate is definitely higher. On Android 9 I was never ever able to download anything (even though using the same application version: 3.5.1).

**Comment by tobiasKaminsky** [comment]
> Sure thing, I sent the mail containing the info.

I logged into testaccount and download Nextcloud Flyer.pdf and the AEG manual.
Both work fine with emulator and android 9.0.

**Comment by levaidaniel** [comment]
Thanks for trying it out!

Is there anything in the logs I pasted here that could point to an error that might cause this?

**Comment by levaidaniel** [comment]
Or better yet, can I enable something in the Android app that would log what goes wrong when I try to download something?

**Comment by tobiasKaminsky** [comment]
The log you posted did not helped, unfortunately…

You can try to provide locgat infos: https://github.com/nextcloud/android/blob/master/README.md#getting-debug-info-via-logcat

**Comment by fmikker** [comment]
Experiencing the same issue. 

Do you need a additional test account?

Platform: CentOS Linux release 7.6.1810 (Core
HTTP-server: httpd24-httpd-2.4.34-7.el7.x86_64
PHP: PHP 7.0.27
